src/user/user-highlights.service.ts

```diff
@@ -96,11 +98,66 @@ export class UserHighlightsService {
     return new PageDto(entities, pageMetaDto);
   }
 
+  async findAllFeatured(pageOptionsDto: PageOptionsDto): Promise<PageDto<DbUserHighlight>> {
+    const queryBuilder = this.baseQueryBuilder();
+
+    queryBuilder.where(`user_highlights.featured = true`).orderBy("user_highlights.updated_at", "DESC");
+    queryBuilder.offset(pageOptionsDto.skip).limit(pageOptionsDto.limit);
+    const itemCount = await queryBuilder.getCount();
+    const entities = await queryBuilder.getMany();
+
+    const pageMetaDto = new PageMetaDto({ itemCount, pageOptionsDto });
+
+    return new PageDto(entities, pageMetaDto);
+  }
+
+  async addFeatured(highlightId: number, userId: number): Promise<DbUserHighlight> {
+    const user = await this.userService.findOneById(userId);
+
+    if (user.role < 100) {
+      throw new ConflictException("You are not authorized to perform this action");
```

Comment:
This should be an `UnauthorizedException` which will return a 401 HTTP status code.

```suggestion
      throw new UnauthorizedException("You are not authorized to perform this action");
```